examples/lux_core_engine/Main.cc

```diff
@@ -0,0 +1,184 @@
+  lightMaterial->SetDiffuse(1.0, 1.0, 1.0);
+
+  VisualPtr ceilingLight = _scene->CreateVisual();
+  ceiling->AddGeometry(_scene->CreatePlane());
+  ceiling->SetLocalScale(2.5, 2.5, 1);
+  ceiling->SetLocalRotation(0, 0, 0);
+  ceiling->SetLocalPosition(0, 0, 2.5);
+  ceiling->SetMaterial(lightMaterial);
+
+  MaterialPtr boxMaterial1 = _scene->CreateMaterial();
+  boxMaterial1->SetDiffuse(1.0, 1.0, 1.0);
+
+  MaterialPtr boxMaterial2 = _scene->CreateMaterial();
+  boxMaterial2->SetDiffuse(1.0, 1.0, 1.0);
+
+  VisualPtr box1 = _scene->CreateVisual();
+  box1->AddGeometry(_scene->CreateBox());
+  box1->SetLocalPosition(0, -0.85, -1.25);
+  box1->SetLocalRotation(0, 0, -IGN_PI / 4);
+  box1->SetLocalScale(1.5, 1.5, 1.5);
+  box1->SetMaterial(boxMaterial1);
+
+  VisualPtr box2 = _scene->CreateVisual();
+  box2->AddGeometry(_scene->CreateBox());
+  box2->SetLocalPosition(0.5, 0.5, -0.5);
+  box2->SetLocalRotation(0, 0, -IGN_PI / 10);
+  box2->SetLocalScale(1.5, 1.5, 3.0);
+  box2->SetMaterial(boxMaterial2);
+
+  // Duck Scene
+  MaterialPtr matte = _scene->CreateMaterial();
+  matte->SetDiffuse(1.0, 0.0, 0.0);
+
+  VisualPtr mesh = _scene->CreateVisual();
+  MeshDescriptor descriptor;
+  descriptor.meshName = "media/duck.dae";
+  common::MeshManager *meshManager = common::MeshManager::Instance();
+  descriptor.mesh = meshManager->Load(descriptor.meshName);
+  MeshPtr meshGeom = _scene->CreateMesh(descriptor);
+  mesh->AddGeometry(meshGeom);
+  mesh->SetLocalRotation(IGN_PI / 2, 0, -IGN_PI / 4);
+  mesh->SetLocalPosition(-0.25, -1.25, 1.25);
+  mesh->SetMaterial(matte);
+}
+
+//////////////////////////////////////////////////
+CameraPtr createCamera(const std::string &_engineName) {
+  // create and populate scene
+  RenderEngine *engine = rendering::engine(_engineName);
+  if (!engine) {
+    std::cout << "Engine '" << _engineName << "' is not supported" << std::endl;
+    return CameraPtr();
+  }
+  ScenePtr scene = engine->CreateScene("scene");
+  buildScene(scene);
+
+  // return camera sensor
+  SensorPtr sensor = scene->SensorByName("camera");
+  return std::dynamic_pointer_cast<Camera>(sensor);
+}
+
+//////////////////////////////////////////////////
+int main(int _argc, char **_argv) {
+  glutInit(&_argc, _argv);
+
+  common::Console::SetVerbosity(4);
+  std::vector<std::string> engineNames;
+  std::vector<CameraPtr> cameras;
+
+  std::cout << "Hint: Make sure to set IGN_RENDERING_PLUGIN_PATH to the " <<
+               "directory containing \'libLuxCoreEngine.so\'" << std::endl;
+
+  std::cout << "It may take a while to initialize LuxCore" << std::endl;
+
+  engineNames.push_back("LuxCoreEngine");
+
+  for (auto engineName : engineNames) {
+    try {
+      CameraPtr camera = createCamera(engineName);
+      if (camera) {
+        cameras.push_back(camera);
+      }
+    } catch (std::exception &ex) {
+      std::cout << ex.what() << std::endl;
+      // std::cerr << "Error starting up: " << engineName << std::endl;
+    }
+  }
```

Comment:
```suggestion
  for (auto engineName : engineNames)
  {
    try {
      CameraPtr camera = createCamera(engineName);
      if (camera)
      {
        cameras.push_back(camera);
      }
    } catch (std::exception &ex) {
      std::cout << ex.what() << std::endl;
    }
  }
```

Reply:
I made this change in the new PR but somehow unable to resolve this conversation here. Github gives me an error when I try to 
![image](https://user-images.githubusercontent.com/28096692/192577627-b2c5bccd-b660-4629-bbb9-ab30bba711e3.png)